catalog/app/components/JsonEditor/State.js

```diff
@@ -82,14 +79,16 @@ function getColumn(obj, columnPath, sortOrder, schema) {
     R.pathOr({}, schemaPath.concat('properties'), schema),
   )
 
+  const sortFunc = R.sort((a, b) => a.sortIndex - b.sortIndex)
```

Comment:
fyi this can be expressed as `R.sortBy(R.prop('sortIndex'))` or `R.sort(R.ascend(R.prop('sortIndex')))`